File: CookMateAI/test-gemini.py
# Python script to test Google Gemini API
import os
import requests
import json
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

# Get API key
api_key = os.getenv('GEMINI_API_KEY')
print(f"API Key available: {bool(api_key)}")
print(f"API Key length: {len(api_key) if api_key else 0}")

# Test direct API access
url = f"https://generativelanguage.googleapis.com/v1/models/gemini-pro:generateContent?key={api_key}"
headers = {"Content-Type": "application/json"}
data = {
    "contents": [
        {
            "parts": [
                {"text": "Write a short greeting."}
            ]
        }
    ]
}

print("Making API request...")
try:
    response = requests.post(url, headers=headers, json=data)
    print(f"Status code: {response.status_code}")
    print(f"Response: {response.text}")
    
    if response.status_code == 200:
        print("API access successful!")
    else:
        print("API access failed.")
except Exception as e:
    print(f"Error: {e}")

File: CookMateAI/test-gemini.js
// Test script to verify Gemini API access
require('dotenv').config();

const { GoogleGenerativeAI } = require('@google/generative-ai');

// Access the API key from environment variables
const apiKey = process.env.GEMINI_API_KEY;

// Function to test API connection
async function testGeminiAPI() {
  try {
    console.log('Testing Gemini API access...');
    console.log('API Key available:', !!apiKey);
    
    if (!apiKey) {
      console.error('No API key found in environment variables');
      return;
    }

    // Initialize the API
    const genAI = new GoogleGenerativeAI(apiKey);
    
    // Try the original model name
    console.log('Trying gemini-pro model...');
    const model = genAI.getGenerativeModel({ model: 'gemini-pro' });
    
    // Generate a simple response
    const prompt = 'Write a one-sentence greeting.';
    console.log('Sending test prompt to API:', prompt);
    
    const result = await model.generateContent(prompt);
    const response = result.response.text();
    
    console.log('API Response received:');
    console.log(response);
    console.log('Gemini API test successful!');
  } catch (error) {
    console.error('Error testing Gemini API:');
    console.error(error.message);
    console.error('Full error:', error);
  }
}

// Run the test
testGeminiAPI();